ingestion-sink/src/main/java/com/mozilla/telemetry/ingestion/sink/transform/PubsubMessageToObjectNode.java

```diff
@@ -265,7 +265,7 @@ private void transformForBqSchema(ObjectNode parent, List<Field> bqFields,
       final Map<String, Field> bqFieldMap = bqFields.stream()
           .collect(Collectors.toMap(Field::getName, Function.identity()));
 
-      for (String jsonFieldName : Sets.newHashSet(parent.fieldNames())) {
+      for (String jsonFieldName : Lists.newArrayList(parent.fieldNames())) {
         final JsonNode value = parent.get(jsonFieldName);
```

Comment:
a crash frame like `{"ip": "0xf08425a","ip\u0000��": null,...}` was handling the `ip\u0000��` field first, which normalized the field name to `ip`, then removed the `ip` from `parent`, then when it handled `ip` would get `null`. This is fixed by always removing `jsonFieldName`, rather than `bqFieldName`.

Reply:
Do we need to make similar adjustments to `PubsubMessageToTableRow` to support the batch backfill case?

Reply:
I'll start a follow-up PR to move `PubsubMessageToObjectNode` to `ingestion-core` and replace `PubsubMessageToTableRow`